lib/src/cli_arguments.dart

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:charcode/charcode.dart';
+import 'package:glob/glob.dart';
+import 'package:glob/list_local_fs.dart';
+import 'package:path/path.dart' as p;
+import 'package:string_scanner/string_scanner.dart';
+
+/// CLI arguments parsed from a `executableAndArgs` string.
+class CliArguments {
+  /// The executable to run.
+  final String executable;
+
+  /// The arguments to the executable, with globs not yet resolved.
+  final List<_Argument> _arguments;
+
+  /// Parses [argString], a shell-style string of space-separated arguments,
+  /// into a list of separate arguments.
+  ///
+  /// If [glob] is `true`, this will parse arguments (outside of quotes) as
+  /// [Glob]s and resolve those globs when [arguments] is called.
+  ///
+  /// Throws a [FormatException] if [argString] is malformed.
+  factory CliArguments.parse(String argString, {bool? glob}) {
+    /// Don't expand globs on Windows by default, since on Windows the
+    /// convention is to let each program handle its own glob expansion.
```

Comment:
This comment about the default value for `glob` should probably be in the doc comment rather than inline.